src/Ocelot/Requester/MessageInvokerPool.cs

```diff
@@ -10,13 +12,18 @@ public class MessageInvokerPool : IMessageInvokerPool
     private readonly IDelegatingHandlerHandlerFactory _handlerFactory;
     private readonly IOcelotLogger _logger;
 
-    public MessageInvokerPool(IDelegatingHandlerHandlerFactory handlerFactory, IOcelotLoggerFactory loggerFactory)
+    public MessageInvokerPool(
+        IDelegatingHandlerHandlerFactory handlerFactory,
+        IOcelotLoggerFactory loggerFactory,
+        IOptions<FileConfiguration> options)
```

Comment:
`IOptions<FileConfiguration> options`
Good approach to inject configuration model using Options pattern.
But why do you inject whole configuration if you need global value only?
Seems the code doesn't reuse other config options, so entire configuration is not required.

Reply:
I did received the only FileGlobalConfiguration, but registered in DI?
How can I better get File GlobalConfiguration?

Reply:
After conducting some code research, I realized my mistake; the issue was not as significant as I thought. It's possible to inject the entire configuration object. The issue with `IOptions<T>` is that it's not utilized across all service classes currently, which indicates a design and development oversight.

`IOptions<FileConfiguration>` can be injected because we bind this `IConfiguration` section in `OcelotBuilder` at [line 55](https://github.com/ThreeMammals/Ocelot/blob/develop/src/Ocelot/DependencyInjection/OcelotBuilder.cs#L55).

Alternatively, binding a new global configuration section would be preferable.

In conclusion, it's feasible to inject both `IConfiguration` and `IOptions<FileConfiguration>` at this point.
But having injected `IOptions<FileGlobalConfiguration>` is better.